### ISSUE TITLE
Things We need to do to improve the site speed

### ISSUE THREAD

**Issue by vicky002** [fossasia/gci15.fossasia.org#367]
- [x] Serve Scaled Images. - Properly sizing images can save many bytes of data. 
  Take an example of this image : http://gci15.fossasia.org/img/gallery/math.png  is resized in HTML or CSS from 1156x1102 to 240x240
- [x] Optimize Images - 
  - Reduce the load times of pages by loading appropriately sized images.
  - Reduce file sizes based on where images will be displayed
  - Resize image files themselves instead of via CSS
  - Save files in appropriate format depending on usage
- [x] Remove query strings from static resources
  - Most proxies, most notably Squid up through version 3.0, do not cache resources with a "?" in their URL even if a Cache-control: public header is present in the response. To enable proxy caching for these resources, remove query strings from references to static resources, and instead encode the parameters into the file names themselves.
- [x] Following needs to be fixed. Giving 404 error: https://pbs.twimg.com/profile_images/591156696438833152/N2NHBXHm_bigger.jpg 
- [x] The following external CSS files were included after an external JavaScript file . To ensure CSS files are downloaded in parallel, we should always include external CSS before external JavaScript.
  http://gci15.fossasia.org/stylesheets/loklak.css

I'm optimiziing images and fixing few small issues. Let me know if anyone is already working on this.

**Comment by jigyasa-grover** [comment]
@vicky002 Thank for your valuable input.

**Comment by vicky002** [comment]
@jig08 Should I start working on it? No one is working on optimizing images right?

**Comment by vicky002** [comment]
Optimized around 11 images of gallery section. Student's images still need to be optimized. 

See : https://github.com/fossasia/gci15.fossasia.org/commit/5bd9fbcd4fabbce267c12e86e9582287f18c1d70

**Comment by tansanrao** [comment]
According to a Site Speed Test, the Search Query to loklak's API has a ~10second wait before loading, causing slow downs. I used the site [http://tools.pingdom.com/fpt/](http://tools.pingdom.com/fpt/) for the tests.

**Comment by vicky002** [comment]
The last issue of loklak.css is fixed in this (https://github.com/Yathannsh/gci15.fossasia.org/commit/f17084611cb5f05e5daa22fd8bf6e28e127f417a) commit. 
Marking it as done. 
I'm working all other fixes, @jig08 can you assign this issue to me?

**Comment by jigyasa-grover** [comment]
Sure, Go ahead @vicky002  !

**Comment by vicky002** [comment]
I optimized few more images see the commit : https://github.com/fossasia/gci15.fossasia.org/pull/464

Only two more tasks are left! :smile:

**Comment by jigyasa-grover** [comment]
Awesome, Send in a PR when you are done with it :)

Edit: Just saw this PR #464 .

**Comment by vicky002** [comment]
I sent a PR check once :)

**Comment by yasoob** [comment]
- [x] We now serve scaled images. 

Moreover, I don't think we can remove the query strings from static resources as most of those resources require them to work properly. What are your suggestions @vicky002 & @jig08 ?

**Comment by mananwason** [comment]
@yasoob Let's keep this open for now. We'll see what happens the query strings are removed. I think it should work but I'm not sure. @vicky002 views?

**Comment by vicky002** [comment]
Caching of Static Resources like CSS and JavaScript plays an important role in page loading time. Query Strings prevents Caching of Static Resources on Proxy servers and on browsers. By Removing Query Strings from Static Resources you can enable caching of static resources and can achieve a significant improvement in page load speed with less page load time.

Let's fix this tomorrow. Before removing the query string, we have to do little testing and we need to write a script to get the latest CSS and JS files.

**Comment by mananwason** [comment]
Awesome @vicky002

**Comment by Juanvulcano** [comment]
@vicky002 @mananwason I would like to work in removing the query strings. 
So far I've only found examples using Wordpress Plugins, do you know any site on github pages that removes query strings? Maybe I could help on that script :+1:

**Comment by vicky002** [comment]
@mananwason @jig08  We can close this issue for now. 
Site has a lot of content and images. We can actually improve site speed just by fixing those images. 

I'm marking this as done for now and creating another issue. We also need to mentioned somewhere about images. Size of the image should be fixed that is width and height (240 x 240) and max size(memory) of the image that can be decided after the discussion may be max 250-500 Kb or 1MB.

**Comment by mananwason** [comment]
Go ahead @vicky002 :+1: .We have an issue #671 for reducing size of images.
